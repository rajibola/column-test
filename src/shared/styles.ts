import {StyleSheet} from 'react-native';
import colors from '../constants/colors';
import {fonts} from '../constants/fonts';
import {wp, hp} from './layout';

export const ListStyles = StyleSheet.create({
  number: {
    marginTop: hp(2),
    ...fonts[14],
    opacity: 0.6,
  },
  name: {
    ...fonts[14],
    alignSelf: 'baseline',
  },
  textContainer: {
    borderBottomWidth: hp(1),
    borderBottomColor: colors.grey,
    paddingVertical: hp(8),
    flex: 1,
  },
  contactContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: wp(20),
    marginBottom: hp(15),
    flex: 1,
  },

  initials: {
    ...fonts[16],
    color: colors.white,
    textTransform: 'uppercase',
  },
  noAvatar: {
    width: wp(45),
    height: wp(45),
    backgroundColor: colors.dark,
    borderRadius: wp(45),
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: wp(10),
  },
});
export const ButtonStyles = StyleSheet.create({
  buttonText: {
    ...fonts[16],
    textTransform: 'uppercase',
  },
  container: {
    height: hp(57),
    paddingHorizontal: wp(36),
    backgroundColor: colors.light,
    position: 'absolute',
    zIndex: 10,
    bottom: hp(26),
    alignSelf: 'center',
    width: wp(342),
    borderRadius: wp(57),
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export const TextStyles = StyleSheet.create({
  RegularText: {
    ...fonts[12],
    fontFamily: 'rubik-regular',
  },

  medium: {
    ...fonts[12],
    fontFamily: 'rubik-medium',
  },
});
