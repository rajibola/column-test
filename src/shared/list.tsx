import React from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import {SharedElement} from 'react-navigation-shared-element';
import {ListItemProps} from '../types/types.d';
import {getInitails} from '../utils/helpers';
import {wp} from './layout';
import {ListStyles as styles} from './styles';
import {MediumText, RegularText} from './text';

export const ListItem = (props: ListItemProps) => {
  const {navigation, item} = props;
  return (
    <TouchableOpacity style={styles.contactContainer} onPress={navigation}>
      <SharedElement id={`item.${item.recordID}.icon`}>
        {item.hasThumbnail ? (
          <Image
            source={{uri: item.thumbnailPath}}
            width={wp(45)}
            height={wp(45)}
            style={styles.noAvatar}
          />
        ) : (
          <View style={styles.noAvatar}>
            <MediumText
              style={styles.initials}
              title={getInitails(item.givenName)}
            />
          </View>
        )}
      </SharedElement>

      <View style={styles.textContainer}>
        <SharedElement id={`item.${item.recordID}.title`}>
          <MediumText
            style={styles.name}
            title={item.givenName + ' ' + item.familyName}
          />
        </SharedElement>
        <RegularText
          style={styles.number}
          title={item?.phoneNumbers[0]?.number}
        />
      </View>
    </TouchableOpacity>
  );
};
