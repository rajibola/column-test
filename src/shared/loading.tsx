import React from 'react';
import {ActivityIndicator, View} from 'react-native';
import colors from '../constants/colors';
import {LoadingProps as Props} from '../types/types.d';
import {LoadingStyles as styles} from './styles';

export const Loading: React.FC<Props> = ({color, bgColor}) => (
  <View style={[styles.container, {backgroundColor: bgColor}]}>
    <ActivityIndicator color={color || colors.dark} size="large" />
  </View>
);
