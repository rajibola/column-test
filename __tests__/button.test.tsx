import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';
import {Text} from 'react-native';
import {Button} from '../src/shared/button';

describe('Button', () => {
  it('Button component renders correctly', () => {
    const text = 'Login';
    const instance = renderer.create(<Button title={text} />);
    expect(instance).toMatchSnapshot();
  });
  it('Button component renders correctly after an update', () => {
    const value1 = 'Hello';
    const value2 = 'Hi';

    const instance = renderer.create(<Button title={value1} />);
    const textInstance = instance.root.findByType(Text);
    expect(textInstance.props.children).toBe(value1);

    instance.update(<Button title={value2} />);
    expect(textInstance.props.children).toBe(value2);
  });

  it('Button responds to onPress', () => {
    const mockFn = jest.fn();

    const {getByText} = render(<Button title="TEST" onPress={mockFn} />);

    fireEvent.press(getByText('TEST'));

    expect(mockFn).toBeCalled();
  });
});
