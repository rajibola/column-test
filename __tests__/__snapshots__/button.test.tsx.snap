// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button component renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#ecf0f1",
      "borderRadius": 103.5,
      "bottom": 47.5,
      "height": 104,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 65,
      "position": "absolute",
      "width": 619.5,
      "zIndex": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#34495e",
          "fontFamily": "rubik-regular",
          "fontSize": 22,
          "lineHeight": 29,
        },
        Object {
          "color": "#34495e",
          "fontSize": 29,
          "lineHeight": 38.5,
          "textTransform": "uppercase",
        },
      ]
    }
  >
    Login
  </Text>
</View>
`;
